### PR TITLE
Update language based on github issue feedback & approval

### LINKED ISSUE #8060: Incorrect statement about passing Nothing as the value of a ParamArray parameter.
In the section titled [Calling a ParamArray](https://docs.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/procedures/parameter-arrays#calling-a-paramarray) we have this statement:

> Nothing — that is, you can omit the ParamArray argument. In this case, an empty array is passed to the procedure. **You can also pass the Nothing keyword, with the same effect.**

The part in bold is incorrect as the following code shows:

```vb
Public Module Module1
  Private Sub foo(ParamArray v() As Object)
    If v Is Nothing Then
      Console.WriteLine("v is null")
    Else
      Console.WriteLine($"v is NOT null and has {v.Length} items.")
    End If
  End Sub

  Public Sub Main()
    foo()
    foo(Nothing)
    Console.ReadLine()
  End Sub
End Module
```

The quoted section correctly says that if you omit the `ParamArray` argument by writing `foo()`, an empty (i.e non-null) array is passed to the procedure. The code above prints `v is NOT null and has 0 items.` to show that this is correct.

The quoted section then **incorrectly says "you can also pass the Nothing keyword, with the same effect"**. The effect is NOT the same because writing `foo(Nothing)` actually passes a NULL array instead of a zero-length/empty/non-null array as was the case when writing `foo()`. The code above prints `v is null` to prove this.

I suggest the quoted section should be modified to read similar to the following:

> Nothing — that is, you can omit the ParamArray argument. In this case, an empty array is passed to the procedure. **NOTE: If you explicitly pass the Nothing keyword, a null array is passed to the procedure and may result in a NullReferenceException if the called procedure does not check for this condition.**

In fact, thinking about the above suggestion, I think there is a ton of code out there that is not expecting people to pass Nothing to a `ParamArray` parameter and would result in a `NullReferenceException` if they did. A quick look into the Rosyln repo shows [this example](https://github.com/dotnet/roslyn/blob/fab7134296816fc80019c60b0f5bef7400cf23ea/src/VisualStudio/Core/Test/ObjectBrowser/Helpers.vb#L82):

```vb
  <Extension>
  Friend Sub VerifyNames(
               list As IVsSimpleObjectList2,
               itemPredicate As Func(Of IVsSimpleObjectList2, UInteger, Boolean),
               ParamArray names As String()
             )

    '<snipped code>

    For i = 0 To names.Length - 1
      'above expression 'names.Length' will throw if Nothing is passed to ParamArray parameter "names".
    Next

    '<snipped code>
  End Sub
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: f3d8f7c7-4da0-8cbe-c1d0-7b6dcc82ddd2
* Version Independent ID: 65957dcf-c867-7893-0bfa-c0d78c3d8935
* Content: [Parameter Arrays (Visual Basic)](https://docs.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/procedures/parameter-arrays)
* Content Source: [docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md](https://github.com/dotnet/docs/blob/master/docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md)
* Product: **dotnet-visualbasic**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

### PR DESCRIPTION
**Pull request by SeanKilleen**
## Summary

Updates the language as suggested & approved in the GH issue.

Fixes https://github.com/dotnet/docs/issues/8060

### CODE REVIEW

File: docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md [hunk 1]
```diff
@@ -28,7 +28,9 @@ Usually, you cannot call a procedure with more arguments than the procedure decl
 ## Calling a ParamArray  
  When you call a procedure that defines a parameter array, you can supply the argument in any one of the following ways:  
   
-- Nothing — that is, you can omit the [ParamArray](../../../../visual-basic/language-reference/modifiers/paramarray.md) argument. In this case, an empty array is passed to the procedure. You can also pass the [Nothing](../../../../visual-basic/language-reference/nothing.md) keyword, with the same effect.  
+- Nothing — that is, you can omit the [ParamArray](../../../../visual-basic/language-reference/modifiers/paramarray.md) argument. In this case, an empty array is passed to the procedure. **NOTE: If you explicitly pass the [Nothing](../../../../visual-basic/language-reference/nothing.md) keyword, a null array is passed to the procedure and may result in a NullReferenceException if the called procedure does not check for this condition.**
```
Comment (Youssef1313) [inline docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md]:
```suggestion
- Nothing — that is, you can omit the [ParamArray](../../../../visual-basic/language-reference/modifiers/paramarray.md) argument. In this case, an empty array is passed to the procedure. 

  > [!NOTE]
  > If you explicitly pass the [Nothing](../../../../visual-basic/language-reference/nothing.md) keyword, a null array is passed to the procedure and may result in a <xref:System.NullReferenceException> if the called procedure does not check for this condition.
```

**Review by Youssef1313 — CHANGES_REQUESTED** [review]
Added a link to NullReferenceException, and added the new note as an alert-style note.

File: docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md [hunk 2]
```diff
@@ -28,7 +28,9 @@ Usually, you cannot call a procedure with more arguments than the procedure decl
 ## Calling a ParamArray  
  When you call a procedure that defines a parameter array, you can supply the argument in any one of the following ways:  
   
-- Nothing — that is, you can omit the [ParamArray](../../../../visual-basic/language-reference/modifiers/paramarray.md) argument. In this case, an empty array is passed to the procedure. You can also pass the [Nothing](../../../../visual-basic/language-reference/nothing.md) keyword, with the same effect.  
+- Nothing — that is, you can omit the [ParamArray](../../../../visual-basic/language-reference/modifiers/paramarray.md) argument. In this case, an empty array is passed to the procedure. **NOTE: If you explicitly pass the [Nothing](../../../../visual-basic/language-reference/nothing.md) keyword, a null array is passed to the procedure and may result in a NullReferenceException if the called procedure does not check for this condition.**  
+
+
```
Comment (Youssef1313) [inline docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md]:
```suggestion
```

File: docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md [hunk 3]
```diff
@@ -28,7 +28,9 @@ Usually, you cannot call a procedure with more arguments than the procedure decl
 ## Calling a ParamArray  
  When you call a procedure that defines a parameter array, you can supply the argument in any one of the following ways:  
   
-- Nothing — that is, you can omit the [ParamArray](../../../../visual-basic/language-reference/modifiers/paramarray.md) argument. In this case, an empty array is passed to the procedure. You can also pass the [Nothing](../../../../visual-basic/language-reference/nothing.md) keyword, with the same effect.  
+- Nothing — that is, you can omit the [ParamArray](../../../../visual-basic/language-reference/modifiers/paramarray.md) argument. In this case, an empty array is passed to the procedure. **NOTE: If you explicitly pass the [Nothing](../../../../visual-basic/language-reference/nothing.md) keyword, a null array is passed to the procedure and may result in a NullReferenceException if the called procedure does not check for this condition.**  
+
```
Comment (Youssef1313) [inline docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md]:
```suggestion
```

**Review by Youssef1313 — CHANGES_REQUESTED** [review]
Markdown lint shows these errors:

```
docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md:33 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md:34 MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 3]
```

This suggestion should fix them.

File: docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md (hunk 1 above)
Comment (SeanKilleen) [reply docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md]: My change was in keeping with the suggestion in the original issue, which had already been approved, so I'd like someone else to weigh in before accepting this (I don't mind it).

File: docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md [hunk 4]
```diff
@@ -28,7 +28,7 @@ Usually, you cannot call a procedure with more arguments than the procedure decl
 ## Calling a ParamArray  
  When you call a procedure that defines a parameter array, you can supply the argument in any one of the following ways:  
   
-- Nothing — that is, you can omit the [ParamArray](../../../../visual-basic/language-reference/modifiers/paramarray.md) argument. In this case, an empty array is passed to the procedure. You can also pass the [Nothing](../../../../visual-basic/language-reference/nothing.md) keyword, with the same effect.  
+- Nothing — that is, you can omit the [ParamArray](../../../../visual-basic/language-reference/modifiers/paramarray.md) argument. In this case, an empty array is passed to the procedure. **NOTE: If you explicitly pass the [Nothing](../../../../visual-basic/language-reference/nothing.md) keyword, a null array is passed to the procedure and may result in a NullReferenceException if the called procedure does not check for this condition.**
```
Comment (BillWagner) [inline docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md]: I'm going to make a slightly different suggestion. The sentence on explicitly passing `Nothing` should be inline with the text, but adding the "note" word would indicate that we should apply the "note" style:

```suggestion
- Nothing — that is, you can omit the [ParamArray](../../../../visual-basic/language-reference/modifiers/paramarray.md) argument. In this case, an empty array is passed to the procedure. If you explicitly pass the [Nothing](../../../../visual-basic/language-reference/nothing.md) keyword, a null array is passed to the procedure and may result in a NullReferenceException if the called procedure does not check for this condition.
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks for making this fix @SeanKilleen , and for the review @Youssef1313 

I'm making a different suggestion to make a slight modification to the wording.

Once that's been accepted, I'll :shipit:.

EDIT: fix autocomplete to incorrect PR author. 😥

File: docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md (hunk 1 above)
Comment (SeanKilleen) [reply docs/visual-basic/programming-guide/language-features/procedures/parameter-arrays.md]: Resolving this in favor of @BillWagner 's suggestion.

**Comment by SeanKilleen** [comment]
@BillWagner ✔️

**Review by BillWagner — APPROVED** [review]

**Comment by SeanFeldman** [comment]
Fix by @SeanKilleen, not me 🙂

**Pull request merged**